editor/test/integration/test_util.clj

```diff
@@ -625,37 +623,66 @@
                     (:k game-object/GameObjectNode)))
     go-id))
 
+(defn- created-node [select-fn-call-logger]
+  (let [calls (call-logger-calls select-fn-call-logger)
+        args (last calls)
+        selection (first args)
+        node-id (first selection)]
+    node-id))
+
 (defn add-referenced-component!
-  "Adds a new instance of a referenced component to the specified
-  game object node inside a transaction and makes it the current
-  selection. Returns the id of the added ReferencedComponent node."
-  [app-view resource go-id]
-  (let [go-id (to-game-object-id go-id)
-        [select-fn selection-volatile] (make-app-view-select-fn app-view)]
-    (game-object/add-component-file go-id resource select-fn)
-    (first @selection-volatile)))
+  "Adds a new instance of a referenced component to the specified game object
+  node. Returns the id of the added ReferencedComponent node."
+  [game-object-or-instance-id component-resource]
+  (let [select-fn (make-call-logger)
+        game-object-id (to-game-object-id game-object-or-instance-id)]
+    (game-object/add-component-file game-object-id component-resource select-fn)
+    (created-node select-fn)))
 
 (defn add-embedded-component!
-  "Adds a new instance of an embedded component to the specified
-  game object node inside a transaction and makes it the current
-  selection. Returns the id of the added EmbeddedComponent node."
-  [app-view resource-type go-id]
-  (let [go-id (to-game-object-id go-id)
-        [select-fn selection-volatile] (make-app-view-select-fn app-view)]
-    (game-object/add-embedded-component-handler {:_node-id go-id :resource-type resource-type} select-fn)
-    (first @selection-volatile)))
+  "Adds a new instance of an embedded component to the specified game object
+  node. Returns the id of the added EmbeddedComponent node."
+  [game-object-or-instance-id resource-type]
+  (let [select-fn (make-call-logger)
+        game-object-id (to-game-object-id game-object-or-instance-id)]
+    (game-object/add-embedded-component-handler {:_node-id game-object-id :resource-type resource-type} select-fn)
+    (created-node select-fn)))
+
+(defn add-referenced-game-object!
+  "Adds a new instance of a referenced game object to the specified collection
+  node. Returns the id of the added ReferencedGOInstanceNode."
+  ([collection-id game-object-resource]
+   (add-referenced-game-object! collection-id collection-id game-object-resource))
+  ([collection-id parent-id game-object-resource]
+   (let [select-fn (make-call-logger)]
+     (collection/add-game-object-file collection-id parent-id game-object-resource select-fn)
+     (created-node select-fn))))
```

Comment:
Added `add-referenced-game-object!` and `add-referenced-collection!` below to complete the set!